impl/src/bitfield_specifier.rs

```diff
@@ -45,17 +45,69 @@ fn generate_or_error(input: TokenStream2) -> syn::Result<TokenStream2> {
     }
 }
 
-fn generate_enum(input: syn::ItemEnum) -> syn::Result<TokenStream2> {
+struct Attributes {
+    bits: Option<usize>,
+}
+
+fn parse_attrs(attrs: &[syn::Attribute]) -> syn::Result<Attributes> {
+    let attributes = attrs
+        .iter()
+        .filter(|attr| attr.path.is_ident("bits"))
+        .fold(
+            Ok(Attributes { bits: None }),
+            |acc: syn::Result<Attributes>, attr| {
+                let mut acc = acc?;
+                if acc.bits.is_some() {
+                    return Err(syn::Error::new(
+                        Span2::call_site(),
+                        "More than one 'bits' attributes is not permitted",
+                    ))
```

Comment:
Please make use of the format_err_spanned macro here. All errors in the proc. macros are handled through them. Also please use some proper span and always avoid using Span2::call_site(). A proper span here is probably the invalid attribute.